### PR TITLE
Fix scaling and frame size

### PR DESCRIPTION
**Pull request by WalterCouto**
This should fix the open issue https://github.com/daidodo/diablo_edit/issues/49, https://github.com/daidodo/diablo_edit/issues/26 and https://github.com/daidodo/diablo_edit/issues/11.

### CODE REVIEW

File: Diablo Edit2/DlgCharItems.cpp [hunk 1]
```diff
@@ -662,14 +663,40 @@ static void DrawGrid(CPaintDC & dc, const CRect & rect, int intervalX = 0, int i
 	}
 }
 
+void CDlgCharItems::CalculateScale()
+{
+	auto pWnd = GetDlgItem(IDC_CHECK1);	// This control should have an pixel left position of 338 and bottom position of 58
```
Comment (daidodo) [inline Diablo Edit2/DlgCharItems.cpp:668]: Where do 338 and 58 come from?

File: Diablo Edit2/DlgCharItems.cpp [hunk 2]
```diff
@@ -162,7 +167,7 @@ const int POSITION_INFO[POSITION_END][5] = {
 static CRect PositionToRect(EPosition pos) {
 	ASSERT(pos < POSITION_END);
 	auto & p = POSITION_INFO[pos];
-	return CRect(p[0], p[1], p[0] + GRID_WIDTH * p[2], p[1] + GRID_WIDTH * p[3]);
+	return CRect(p[0] * DLG_SCALE_X, p[1] * DLG_SCALE_Y, p[0] * DLG_SCALE_X + GRID_WIDTH_SCALED * p[2], p[1] * DLG_SCALE_Y + GRID_WIDTH_SCALED * p[3]);
```
Comment (daidodo) [inline Diablo Edit2/DlgCharItems.cpp:170]: `C4244: conversion from 'double' to 'int', possible loss of data`

**Comment by daidodo** [comment]
Thanks very much for the PR!

Could you give some details on how it will solve the layout issue?

File: Diablo Edit2/DlgCharItems.cpp (hunk 1 above)
Comment (WalterCouto) [reply Diablo Edit2/DlgCharItems.cpp:668]: There is no RECT defining the drawing area for the different grids, so you can get the scale of the grid by the controls on the page.  The IDC_CHECK! control is in a good spot to determine what the scale is for both width and height.  The gird is supposed to be 30 x 30 pixels, and  all grids or equipped slots are a multiple of that.  So, you can calculate what the position of that control should have been to fit those controls without scaling (i.e., scale of 1.0).  This is where this number comes from.

File: Diablo Edit2/DlgCharItems.cpp (hunk 2 above)
Comment (WalterCouto) [reply Diablo Edit2/DlgCharItems.cpp:170]: I missed this warning with other similar warnings in compiling, but the warning is not really valid for here as we are taking the floor of that operation always.  So we can add (LONG) to the multiplication.

**Pull request merged**